pkgs/tools/filesystems/fatresize/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchurl, parted, substituteAll, utillinux, ... }:
+
+stdenv.mkDerivation rec {
+  
+  version = "1.0.2";
+  name = "fatresize-${version}";
+
+  src = fetchurl {
+    url = mirror://sourceforge/fatresize/fatresize-1.0.2.tar.bz2;
+    sha256 = "04wp48jpdvq4nn0dgbw5za07g842rnxlh9mig4mslz70zqs0izjm";
+  };
+  
+  phases = [
+    "unpackPhase"
+    "patchPhase"
+    "configurePhase"
+    "buildPhase"
+    "installPhase"
+  ];
+  
+  buildInputs = [ parted utillinux ];
+  
+  patches = [ ./parted_nix.patch ];
+  
+  preConfigure = ''
+    echo "Replacing calls to ped_free with free ..."
+    substituteInPlace ./fatresize.c --replace ped_free free
+  '';
+  
+  makeFlags = ''
+    LDFLAGS=-lparted-fs-resize
+  '';
+  
+  propagatedBuildInputs = [ parted utillinux ];
+  
+  meta = {
+    description = "The FAT16/FAT32 non-destructive resizer.";
```

Comment:
Hi,

I have read your mails - sorry for the late reply ...
I am starting to correct the issues you've pointed out :)
Thanks for your support

2015-08-18 1:02 GMT+03:00 Tobias Geerinckx-Rice <notifications@github.com>:

> In pkgs/tools/filesystems/fatresize/default.nix
> <https://github.com/NixOS/nixpkgs/pull/9301#discussion_r37242369>:
>
> > +
> > +  patches = [ ./parted_nix.patch ];
> > +
> > +  preConfigure = ''
> > +    echo "Replacing calls to ped_free with free ..."
> > +    substituteInPlace ./fatresize.c --replace ped_free free
> > +  '';
> > +
> > +  makeFlags = ''
> > +    LDFLAGS=-lparted-fs-resize
> > +  '';
> > +
> > +  propagatedBuildInputs = [ parted utillinux ];
> > +
> > +  meta = {
> > +    description = "The FAT16/FAT32 non-destructive resizer.";
>
> The full stop is unnecessary. See
> http://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/NixOS/nixpkgs/pull/9301/files#r37242369>.
>